File: TOMIMSV4/Entity/Mantenimientos/Empresa/clsBeEmpresa.vb
<Serializable>
Public Class clsBeEmpresa : Inherits clsBeEmpresaBase
    Implements ICloneable
    Implements IDisposable

    'Public Property IdEmpresa() As Integer = 0
    'Public Property Nombre() As String = ""
    Public Property Direccion() As String = ""
    Public Property Telefono() As String = ""
    Public Property Email() As String = ""
    Public Property Razon_social() As String = ""
    Public Property Representante() As String = ""
    Public Property Corr_cod_barra() As Integer = 0
    Public Property Path_printer() As String = ""
    Public Property Activo() As Boolean = False
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property ClienteRapido() As Boolean = False
    'Public Overloads Property Imagen() As Byte()
    Public Property Operador_logistico() As Boolean = False
    Public Property Puerto_escaner() As Integer = 0
    Public Property Control_presentaciones() As Boolean = False
    Public Property Anulaciones_por_supervisor() As Boolean = False
    Public Property Codigo() As String = ""
    Public Property Clave() As String = ""
    Public Property Intento() As Integer = 0
    Public Property Duracionclave() As Integer = 0
    Public Property Duracionclavetemporal() As Integer = 0
    Public Property codigo_automatico As Boolean = False
    Public Property politica_contraseñas As Boolean = False
    Public Property IdMotivoAjusteInventario() As Integer = 0

    ''' <summary>
    ''' '#CKFK 20211006 Corregí el Año porque estaba como 1999
    ''' </summary>
    ''' <returns></returns>
    Public Property Hora_Corte_Jornada_Sistema As Date = New Date(1900, 1, 1, 23, 59, 59)
    Public Property Generar_Stock_Jornada As Boolean = False

    ''' <summary>
    ''' #EJC20220802: Para control de versión de la base de datos.
    ''' </summary>
    ''' <returns></returns>
    Public Property Version_BD As String = "0"

    ''' <summary>
    ''' #EJC20220802: Para identificarse en la base de datos de amazon (del futuro)
    ''' </summary>
    ''' <returns></returns>
    Public Property AWS_Token As String = ""

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

#Region "IDisposable Support"
    Private disposedValue As Boolean ' To detect redundant calls

    ' IDisposable
    Protected Overridable Sub Dispose(disposing As Boolean)
        If Not disposedValue Then
            If disposing Then
                ' TODO: dispose managed state (managed objects).
            End If

            ' TODO: free unmanaged resources (unmanaged objects) and override Finalize() below.
            ' TODO: set large fields to null.
        End If
        disposedValue = True
    End Sub

    ' TODO: override Finalize() only if Dispose(disposing As Boolean) above has code to free unmanaged resources.
    'Protected Overrides Sub Finalize()
    '    ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
    '    Dispose(False)
    '    MyBase.Finalize()
    'End Sub

    ' This code added by Visual Basic to correctly implement the disposable pattern.
    Public Sub Dispose() Implements IDisposable.Dispose
        ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
        Dispose(True)
        ' TODO: uncomment the following line if Finalize() is overridden above.
        ' GC.SuppressFinalize(Me)
    End Sub
#End Region

End Class

<Serializable>
Public Class clsBeEmpresaBase

    Public Property IdEmpresa() As Integer = 0
    Public Property Nombre() As String = ""
    Public Property Imagen As Byte()

    '#GT12052022: si la app recibe o no updates (cealsa)
    Public Property buscar_actualizacion_hh As Boolean = False

End Class

File: WMS.EntityCore/Datos_Maestros/clsBeEmpresa.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMSWebAPI.Be
{
    public class clsBeEmpresa : ICloneable
    {
        [Column("IdEmpresa")]
        [DisplayName("IdEmpresa")]
        public int IdEmpresa { get; set; } = 0;

        [Column("nombre")]
        [DisplayName("nombre")]
        public string Nombre { get; set; } = "";

        [Column("direccion")]
        [DisplayName("direccion")]
        public string Direccion { get; set; } = "";

        [Column("telefono")]
        [DisplayName("telefono")]
        public string Telefono { get; set; } = "";

        [Column("email")]
        [DisplayName("email")]
        public string Email { get; set; } = "";

        [Column("razon_social")]
        [DisplayName("razon_social")]
        public string Razon_social { get; set; } = "";

        [Column("representante")]
        [DisplayName("representante")]
        public string Representante { get; set; } = "";

        [Column("corr_cod_barra")]
        [DisplayName("corr_cod_barra")]
        public int Corr_cod_barra { get; set; } = 0;

        [Column("path_printer")]
        [DisplayName("path_printer")]
        public string Path_printer { get; set; } = "";

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("clienteRapido")]
        [DisplayName("clienteRapido")]
        public bool ClienteRapido { get; set; } = false;

        [Column("imagen")]
        [DisplayName("imagen")]
        public byte[] Imagen { get; set; } = Array.Empty<byte>();

        [Column("operador_logistico")]
        [DisplayName("operador_logistico")]
        public bool Operador_logistico { get; set; } = false;

        [Column("puerto_escaner")]
        [DisplayName("puerto_escaner")]
        public int Puerto_escaner { get; set; } = 0;

        [Column("control_presentaciones")]
        [DisplayName("control_presentaciones")]
        public bool Control_presentaciones { get; set; } = false;

        [Column("anulaciones_por_supervisor")]
        [DisplayName("anulaciones_por_supervisor")]
        public bool Anulaciones_por_supervisor { get; set; } = false;

        [Column("codigo")]
        [DisplayName("codigo")]
        public string Codigo { get; set; } = "";

        [Column("clave")]
        [DisplayName("clave")]
        public string Clave { get; set; } = "";

        [Column("intento")]
        [DisplayName("intento")]
        public int Intento { get; set; } = 0;

        [Column("duracionclave")]
        [DisplayName("duracionclave")]
        public int Duracionclave { get; set; } = 0;

        [Column("duracionclavetemporal")]
        [DisplayName("duracionclavetemporal")]
        public int Duracionclavetemporal { get; set; } = 0;

        [Column("codigo_automatico")]
        [DisplayName("codigo_automatico")]
        public bool Codigo_automatico { get; set; } = false;

        [Column("politica_contraseñas")]
        [DisplayName("politica_contraseñas")]
        public bool Politica_contraseñas { get; set; } = false;

        [Column("IdMotivoAjusteInventario")]
        [DisplayName("IdMotivoAjusteInventario")]
        public int IdMotivoAjusteInventario { get; set; } = 0;

        [Column("cantidad_decimales_despliegue")]
        [DisplayName("cantidad_decimales_despliegue")]
        public int Cantidad_decimales_despliegue { get; set; } = 0;

        [Column("cantidad_decimales_calculo")]
        [DisplayName("cantidad_decimales_calculo")]
        public int Cantidad_decimales_calculo { get; set; } = 0;

        [Column("minutos_timer_jornada_sistema")]
        [DisplayName("minutos_timer_jornada_sistema")]
        public double Minutos_timer_jornada_sistema { get; set; } = 0;

        [Column("hora_corte_jornada_sistema")]
        [DisplayName("hora_corte_jornada_sistema")]
        public DateTime Hora_corte_jornada_sistema { get; set; } = DateTime.Now;

        [Column("generar_stock_jornada")]
        [DisplayName("generar_stock_jornada")]
        public bool Generar_stock_jornada { get; set; } = false;

        [Column("buscar_actualizacion_hh")]
        [DisplayName("buscar_actualizacion_hh")]
        public bool Buscar_actualizacion_hh { get; set; } = false;

        [Column("version_bd")]
        [DisplayName("version_bd")]
        public string Version_bd { get; set; } = "";

        [Column("aws_token")]
        [DisplayName("aws_token")]
        public string Aws_token { get; set; } = "";

        public clsBeEmpresa() { }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}